src/main/diagnostics/steps/internal/loggerHooks.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ElectronLog} from 'electron-log';
+
+import {MASK_EMAIL, MASK_IPV4, MASK_PATH, MASK_URL, REGEX_EMAIL, REGEX_IPV4, REGEX_PATH_DARWIN, REGEX_PATH_LINUX, REGEX_PATH_WIN32, REGEX_URL} from 'common/constants';
+
+import {truncateString} from './utils';
+
+type ElectronLogHook = ElectronLog['hooks'][number];
+type ElectronLogHookCreator = (l: ElectronLog) => ElectronLogHook;
+
+const isDarwin = process.platform === 'darwin';
+const isLinux = process.platform === 'linux';
+const isWin = process.platform === 'win32';
+
+export const maskMessageDataHook: ElectronLogHookCreator = (logger) => (message, transport) => {
+    if (transport !== logger.transports.file) {
+        return message;
+    }
+
+    for (let i = 0; i < message.data.length; i++) {
+        // Specific keywords
+        if (message.data[i]?.toLowerCase?.().includes('password')) {
+            return false;
```

Comment:
(question) What's the purpose of returning? to avoid logging altogether in case password is in the log? Why not filtering out the password?